### PR TITLE
Update docker image

### PR DESCRIPTION
**Pull request by Rombur**
relate to #208

### CODE REVIEW

**Comment by aprokop** [comment]
Do we still need `trilinos_kokkos.patch` in Jenkins?

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '2'
 services:
   dtk_dev:
-    image: dalg24/dtk-stack:latest
+    image: dalg24/dtk-stack:17.07
```
Comment (dalg24) [inline docker/docker-compose.yml]: Goes in separate commit and will be dropped before merging.

Also I would vote for YY.MM.0

File: docker/Dockerfile_stack [hunk 2]
```diff
@@ -26,6 +26,9 @@ RUN apt-get update && apt-get install -y \
         libatlas-base-dev \
         zlib1g-dev \
         python2.7-dev \
+        ninja-build \
+        python-sphinx \
+        python-breathe \
```
Comment (dalg24) [inline docker/Dockerfile_stack]: We need specific versions of sphinx and breathe.  We will use pip to install them.

File: docker/Dockerfile_stack [hunk 3]
```diff
@@ -218,16 +223,22 @@ RUN export TRILINOS_VERSION=12.10.1 && \
...
-    mv kokkos kokkos_outdated && \
-    mkdir kokkos && \
-    tar -xf ${KOKKOS_ARCHIVE} -C kokkos --strip-components=1 && \
-    rm -rf ${KOKKOS_ARCHIVE}
+
+# download Kokkos-Tools
+# sh does not support arrays so we need to use bash
+RUN ["/bin/bash","-c","export KOKKOS_TOOLS_HASH=7ee5bcb80a8329c8493da735bc7bad5a859dfb1f && \
+    export KOKKOS_TOOLS_SHORT_HASH=7ee5 && \
+    export KOKKOS_TOOLS_ARRAY=(\"kernel-filter\" \"kernel-logger\" \"memory-events\" \"memory-hwm-mpi\" \"memory-hwm\" \"memory-usage\" \"simple-kernel-timer-json\" \"simple-kernel-timer\" \"space-time-stack\") && \
```
Comment (dalg24) [inline docker/Dockerfile_stack]: Plz break that line for readability.

File: docker/Dockerfile_stack [hunk 4]
```diff
@@ -218,16 +223,22 @@ RUN export TRILINOS_VERSION=12.10.1 && \
...
+    export KOKKOS_TOOLS_URL=https://github.com/kokkos/kokkos-tools/archive/${KOKKOS_TOOLS_HASH}.tar.gz && \
+    export KOKKOS_TOOLS_ARCHIVE=${PREFIX}/archive/kokkos-tools-${KOKKOS_TOOLS_HASH}.tar.xz && \
+    export KOKKOS_TOOLS_SOURCE_DIR=${PREFIX}/source/kokkos-tools/${KOKKOS_TOOLS_SHORT_HASH} && \
+    wget --quiet ${KOKKOS_TOOLS_URL} --output-document=${KOKKOS_TOOLS_ARCHIVE} && \
+    mkdir -p ${KOKKOS_TOOLS_SOURCE_DIR} && \
+    tar -xf ${KOKKOS_TOOLS_ARCHIVE} -C ${KOKKOS_TOOLS_SOURCE_DIR} --strip-components=1 && \
+    cd ${KOKKOS_TOOLS_SOURCE_DIR}/src/tools && \
+    for tool in ${KOKKOS_TOOLS_ARRAY[@]}; do cd $tool && make && cd ../; done && \
+    rm -rf ${KOKKOS_TOOLS_ARCHIVE}"]
+
```
Comment (dalg24) [inline docker/Dockerfile_stack]: Set env variable for kokkos-tools.

**Comment by Rombur** [comment]
On top of Jenkins, I also ran the following tests:
`docker run --rm -it -v $PWD:/scratch -w /scratch dalg24/dtk-stack:17.07 bash -c "sphinx-build docs/source docs/build"`
`docker run --rm -it -v $PWD:/scratch -w /scratch dalg24/dtk-stack:17.07 bash -c "docker/travis/check_format_cpp.sh"`
`docker run --rm -it -v $PWD:/scratch -w /scratch dalg24/dtk-stack:17.07 bash -c "docker/travis/check_no_trailing_whitespaces_or_tabs.sh"`

Note that the last commit needs to be dropped before we merge this.

**Comment by Rombur** [comment]
Ready to merge

**Comment by dalg24** [comment]
I tagged the image `17.07.0` and overwrote 'latest' on Docker Hub.

**Pull request merged**